content/v2.0/application/settings.md

```diff
@@ -0,0 +1,266 @@
+
+If a setting uses a constructor like `Types::String` or `Types::Params::Bool`, then Hanami will raise an exception if the setting is missing, rather than let your application start in a potentially invalid state.
+
+The below settings configuration will result in a `Hanami::Settings::InvalidSettingsError` when `DATABASE_URL` and `ANALYTICS_ENABLED` environment variables are not set:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool
+    setting :max_cart_items, constructor: Types::Params::Integer
+  end
+end
+```
+
+```shell
+bundle exec hanami server
+
+! Unable to load application: Hanami::Settings::InvalidSettingsError: Could not initialize settings. The following settings were invalid:
+
+analytics_enabled:  cannot be coerced to false
+max_cart_items: can’t convert nil into Integer
+```
+
+The same exception will be raised if a setting can't be correctly coerced:
+
+```shell
+ANALYTICS_ENABLED=true MAX_CART_ITEMS="not coerceable to integer"  bundle exec hanami server
+
+! Unable to load application: Hanami::Settings::InvalidSettingsError: Could not initialize settings. The following settings were invalid:
+
+max_cart_items: invalid value for Integer(): "not coerceable to integer"
+```
+
+To make a setting optional, use `optional` to allow `nil` values:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool.optional
+    setting :max_cart_items, constructor: Types::Params::Integer.optional
+  end
+end
+```
+
+## Default values
+
+Settings can also specify defaults to be used in the absence of the relevant environment variable.
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :redis_url, default: "redis://localhost:6379", constructor: Types::String
+    setting :analytics_enabled, default: false, constructor: Types::Params::Bool
+    setting :max_cart_items, default: 100, constructor: Types::Params::Integer
+  end
+end
+```
+
+## Constraints
+
+To enforce additional contraints on settings, you can use a [dry-types](https://dry-rb.org/gems/dry-types) constraint in your constructor.
+
+Here, the value of the `session_secret` must be at least 32 characters, while the value of the `from_email` setting must satisfy the EMAIL_FORMAT regular expression:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    EMAIL_FORMAT = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+
+    setting :session_secret, constructor: Types::String.constrained(min_size: 32)
+    setting :from_email, constructor: Types::String.constrained(format: EMAIL_FORMAT)
+  end
+end
+```
+
+## Using settings within your application
```

Comment:
Perhaps this could be "Using settings within your app and slices", and underneath the example below, we could have an example of how the settings are also available as a `"settings"` component within every slice?

The reason not to do this here is if we planned to have a separate part in the dedicated slices section that covers all the shared components from the app. TBH this might be better, because it helps the other docs like this one remain focused on their main topic.

Reply:
Yeah, I was of the mind that the settings per slice should go in the slices guide. Given the idea that slices are optional and opt in.